lib/vtls/schannel_verify.c

```diff
@@ -491,97 +557,123 @@ CURLcode Curl_verify_host(struct Curl_cfilter *cf,
     char buffer[STRERROR_LEN];
     failf(data, "schannel: Failed to read remote certificate context: %s",
           Curl_sspi_strerror(sspi_status, buffer, sizeof(buffer)));
-    result = CURLE_PEER_FAILED_VERIFICATION;
     goto cleanup;
   }
 
-  /* Determine the size of the string needed for the cert hostname */
-  len = cert_get_name_string(data, pCertContextServer, NULL, 0);
-  if(len == 0) {
-    failf(data,
-          "schannel: CertGetNameString() returned no "
-          "certificate name information");
-    result = CURLE_PEER_FAILED_VERIFICATION;
-    goto cleanup;
+  Win8_compat = curlx_verify_windows_version(6, 2, 0, PLATFORM_WINNT,
+                                             VERSION_GREATER_THAN_EQUAL);
+  if(get_num_host_info(p, conn_hostname) || !Win8_compat) {
```

Comment:
Since `get_num_host_info()` returns TRUE if the hostname is an IP address. Should this condition then not rather check if it is *not* an IP address?

```suggestion
  if(!get_num_host_info(p, conn_hostname) || !Win8_compat) {
```

Reply:
No. The core functionality of the code hasn't changed since day 1, not even a single bit. 100% of the changes are cosmetic, like renaming variables, or defining structures instead of casting raw buffers as structures. Here is how it works:

We need to get the Subject Alternative Name information manually, if:
***
- The operating system is Windows Vista or 7 (!Win8_compat)
**OR**
- We're dealing with numeric IP Addresses (get_num_host_info has returned TRUE)
***
Obviously, both conditions can be TRUE at the same time, which requires further validation. The following code block verifies the validity of the `cbData` (let's call it size) in the `ip_blob`.

If the size isn't 0, it means that we're dealing numeric IP Addresses (OS version becomes irrelevant), and we can proceed with the verification process (the **for** loop).

Otherwise we go with the host name verification process which is OS version dependent, and that's where `Win8_compat` comes into play (long story short, we enter the **else** branch and call the `cert_get_name_string` with Win8_compat as a parameter).